namer/dnssrv/src/main/scala/io/buoyant/namer/dnssrv/DnsSrvNamer.scala

```diff
@@ -58,19 +66,26 @@ class DnsSrvNamer(
     }
   }
 
-  private def watchDns(dsnsrvRecord: String, timer: Timer): Activity[List[Address]] = {
-    val state = Var.async[Activity.State[List[Address]]](Activity.Pending) { update =>
+  private def watchDns(dnsSrvName: String, timer: Timer): Activity[List[Address]] = {
+    val iteration = new AtomicInteger(0)
+    val gauge = dnsIter.addGauge(dnsSrvName)(iteration.floatValue) // emit PER DNS iteration count, ref this val in the doUnit to avoid GC
 
+    val state: Var[Activity.State[List[Address]]] = Var.async[Activity.State[List[Address]]](Activity.Pending) { update =>
       def doUnit(): Unit = {
-        val lookup = new DNS.Lookup(dsnsrvRecord, DNS.Type.SRV, DNS.DClass.IN)
-        lookup.setResolver(resolver)
+        val lookup = new DNS.Lookup(dnsSrvName, DNS.Type.SRV, DNS.DClass.IN)
+        iteration.incrementAndGet()
+        lookup.setResolver(resolver())
```

Comment:
this creates a new resolver in every iteration. Probably ok, since this isn't a hot loop (it's scheduled to run once every couple of seconds, per DNS address), but it's some file I/O nonetheless. Might be worth investigating if we can use one resolver per address instead (e.g. create it outside `doUnit`)

Reply:
We have tested on production for 3 weeks handling many 10k/s req with 3
namerd boxes - they are small instance - so I would think the scalability
is fine :)